### DISCUSSION TITLE
Convert to positive resist entire cell

### DISCUSSION THREAD

**Discussion by nicholasnobile** [HelgeGehring/gdshelpers#63] [Ideas]
It would be cool if there was support for a convert to positive resist that would work with an entire cell, doing the conversion for all shapes for all sub-cells and all cell arrays. currently, there is not a great way to do this that supports both cells in cells and cell arrays of sub-cells.

**Comment by HelgeGehring** [comment]
At the moment my suggestion would be to use get_reduced_layer to get all shapes on a certain layer as one shapely object and to convert this object to positive resist and put in on another layer. Does that help?

**Reply by nicholasnobile** [reply]
Believe it or not, I have already tried this out. This does not work with cell arrays currently....Any time you add a cell with more than one row/column, this causes only the first one to appear in the final design when doing this. Other instances of the cell are simply not present.

**Reply by HelgeGehring** [reply]
It was not yet implemented in that function, I don't use the array-function too much, so it might be ignored at some positions :/ for get_reduced_layer() I fixed it now in 53970dd1383f3a6233dc40250b661990c97f1dc6 :)
